frontend/src/utils/error_handling.js

```diff
@@ -1,34 +1,116 @@
+      }
+    }
+
+    // If not array, then most likely another (recursive) error
+    else {
+      result = result.concat(parse_errors(msg_list_or_error, field));
     }
   }
-  for (const [field, error_message] of Object.entries(response_json)) {
-    if (field == model) continue;
-    errors.push([field, error_message]);
-  }
-  return errors;
+  return result;
 }
 
-function get_user_errors(response_json) {
-  return get_profile_errors("user", response_json);
+// Helper to convert snake_case to Header Case,
+// for prettier error messages.
+// Ripped from https://github.com/huynhsamha/js-convert-case/blob/master/src/modules/js-headercase/index.ts
+// of https://www.npmjs.com/package/js-convert-case
+
+function toHeaderCase(str) {
+  if (!str) return "";
+
+  return String(str)
+    .replace(/^[^A-Za-z0-9]*|[^A-Za-z0-9]*$/g, "")
+    .replace(/([a-z])([A-Z])/g, (m, a, b) => `${a}_${b.toLowerCase()}`)
+    .replace(/[^A-Za-z0-9]+|_+/g, " ")
+    .toLowerCase()
+    .replace(
+      /( ?)(\w+)( ?)/g,
+      (m, a, b, c) => a + b.charAt(0).toUpperCase() + b.slice(1) + c
+    );
 }
 
-function get_team_errors(response_json) {
-  return get_profile_errors("team", response_json);
+function print_errors(xhr) {
+  // For ease of debugging and customer service, dump to console.
+  // (So that Teh Devs can ask users to open console when things go wrong)
+  // This isn't a security hole; all info here is visible from the browser anyways,
+  // sometimes with some extra annoying clicks.
+  console.log("Error:");
+  console.log(xhr);
+  console.log("Error JSON, if exists:");
+  console.log(xhr.responseJSON);
+
+  // We parse errors from the response JSON.
+  const input = xhr.responseJSON;
+
+  // Just in case...
+  if (!input) {
+    return `There was an error. See your browser console or ask Teh Devs for assistance.`;
+  }
+
+  // Most of the time, errors come from serializers, and the errors are large objects.
+  // However, the backend errors can be simple "{detail: ...}"
+  // Handle those.
+  if (input.detail && Object.keys(input).length === 1) {
+    return `Error: ${input.detail}`;
+  }
+
+  // Similar for "non_field_errors"
+  // Note that doing this check for detail, non_field_errors, and many more
+  // can get repetitive...find a DRY way for this if needed later.
+  if (input.non_field_errors && Object.keys(input).length === 1) {
+    return `Error: ${input.non_field_errors}`;
+  }
+
+  const errors = parse_errors(input);
+
+  if (errors.length === 1) {
+    const error = errors[0];
+    return `Error in ${error.field}: ${error.msg}`;
+  } else {
+    return `Errors:\n${errors
+      .map((error) => `In ${error.field}: ${error.msg}`)
+      .join("\n")}`;
+  }
 }
 
-function get_nested_profile_errors(response_json) {
-  return get_profile_errors("profile", response_json);
+// Copy-paste into console for convenient testing.
+// Add more test cases too, within the function, as desired.
+function test() {
```

Comment:
tests should probably be elsewhere, but keeping it here seems like the best option for now. but could you call this something more descriptive, like `test_error_handling`?